include/eve/module/core/regular/impl/simd/riscv/store.hpp

```diff
@@ -0,0 +1,77 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+#include <eve/arch/riscv/rvv_common_masks.hpp>
+#include <eve/concept/value.hpp>
+#include <eve/detail/category.hpp>
+#include <eve/detail/implementation.hpp>
+#include <eve/module/core/regular/unalign.hpp>
+
+namespace eve::detail
+{
+
+template<arithmetic_scalar_value T, typename N, simd_compatible_ptr<wide<T, N>> Ptr>
+EVE_FORCEINLINE void
+riscv_store(logical<wide<T, N>> mask, wide<T, N> v, Ptr p)
+requires rvv_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, N>>();
+  if constexpr( match(c, category::size8_) ) return __riscv_vse8(mask, p, v, N::value);
+  if constexpr( match(c, category::size16_) ) return __riscv_vse16(mask, p, v, N::value);
+  if constexpr( match(c, category::size32_) ) return __riscv_vse32(mask, p, v, N::value);
+  if constexpr( match(c, category::size64_) ) return __riscv_vse64(mask, p, v, N::value);
+}
+
+template<arithmetic_scalar_value T, typename N, simd_compatible_ptr<wide<T, N>> Ptr>
+EVE_FORCEINLINE void
+riscv_store(wide<T, N> v, Ptr p)
+requires rvv_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, N>>();
+  if constexpr( match(c, category::size8_) ) return __riscv_vse8(p, v, N::value);
+  if constexpr( match(c, category::size16_) ) return __riscv_vse16(p, v, N::value);
+  if constexpr( match(c, category::size32_) ) return __riscv_vse32(p, v, N::value);
+  if constexpr( match(c, category::size64_) ) return __riscv_vse64(p, v, N::value);
+}
+
+// Regular store
+template<arithmetic_scalar_value T, typename N, simd_compatible_ptr<wide<T, N>> Ptr>
+EVE_FORCEINLINE void
+store_(EVE_SUPPORTS(rvv_), wide<T, N> v, Ptr p)
+requires(rvv_abi<abi_t<T, N>> && !has_store_equivalent<wide<T, N>, Ptr>)
+{
+  auto const tgt = as<wide<T>> {};
+  auto       ptr = unalign(p);
+
+  return riscv_store(v, p);
+}
+
+// Conditional store
+template<scalar_value T,
+         typename N,
+         relative_conditional_expr       C,
+         simd_compatible_ptr<wide<T, N>> Ptr>
+EVE_FORCEINLINE void
+store_(EVE_SUPPORTS(rvv_), C const& cond, wide<T, N> const& v, Ptr ptr) noexcept
+requires rvv_abi<abi_t<T, N>> && (!has_store_equivalent<wide<T, N>, Ptr>)
+{
+  auto p = unalign(ptr);
+  if constexpr( C::is_complete )
+  {
+    if constexpr( !C::is_inverted ) return;
```

Comment:
!is_inverted but has alternative should write the thing.

https://github.com/jfalcou/eve/blob/c2dacd782c29177ffc4b8ef86a900d6bc6389e67/include/eve/module/core/regular/impl/store.hpp#L119

Did you enable the tests for store?